tests/buildkite/golden/pipeline-C.yml

```diff
@@ -0,0 +1,30 @@
+steps:
+- label: A
+- depends_on:
+  - A
+  label: B
+  skip: skipped due to lack of changes
+- depends_on:
+  - A
+  label: C
+- depends_on:
+  - B
+  label: D
+  skip: skipped due to lack of changes
+- depends_on:
+  - C
+  label: E
+- depends_on:
+  - D
+  label: F
+  skip: skipped due to lack of changes
+- depends_on:
+  - E
+  - D
+  label: G
```

Comment:
:thinking: is this correct?

`E` was triggered to run, but `G` here also depends on `D`, which is `skip`ped.

I see that you mentioned this in the comment as well:
> If step 'C' is triggered to run, then steps 'A', 'E', 'G', 'H' will be triggered to run as well.

but is this what we want?

AFAIU if `G` needs an artifact from `D` then it will fail because of this, right?

Reply:
indeed, I've added a fix to visit `G` and all its dependencies as well